cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -149,6 +178,27 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     return new DataprocClient(new DataprocConf(conf, network, subnet), client, compute, useInternalIP);
   }
 
+  private static PeeringState peeringState(String systemNetwork, String systemProjectId, Network networkInfo) {
+    List<NetworkPeering> peerings = networkInfo.getPeerings();
+    for (NetworkPeering peering : peerings) {
+      Matcher matcher = GCP_SYSTEM_NETWORK_FULL_PATH.matcher(peering.getNetwork());
+      boolean match = matcher.find();
+      if (!match) {
+        continue;
+      }
+
+      String peeredProjectID = matcher.group(2);
+      String peeredNetwork = matcher.group(4);
+      if (peeredProjectID.equals(systemProjectId) && peeredNetwork.equals(systemNetwork)) {
```

Comment:
As mentioned previously, we should not be using pattern matching for this. The Network object for the system network can be fetched and used to check equality. The global location should not be hardcoded either, as it should be known from the Network object.

Reply:
removed the pattern matching code.